.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,77 @@
+name: "Initialize Virtual Env"
+runs:
+  using: composite
+  steps:
+    - name: Restore nvm, pre-commit cache, & node modules from cache
+      id: cache-nvm-node-modules
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.nvm
+          ~/.cache
+          frontend/node_modules
+        key: v13-nvm-node-modules-${{ hashFiles('**/yarn.lock.md5') }}-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    - name: Install nvm, pre-commit cache and node_modules
+      if: steps.cache-nvm-node-modules.cache-hit != 'true'
+      run: |
+        if [ ! -d ~/.nvm ] ; then
+          cd ..
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          mkdir $NVM_DIR
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+          echo "source $NVM_DIR/nvm.sh --install" >> "$HOME/.bash_profile"
+          cd -
+          source $HOME/.bash_profile
+          echo "lts/*" > .nvmrc
+          nvm install
+        fi
+        npm install --location=global yarn
+        if [ ! -d frontend/node_modules ] ; then
+          source "$HOME/.nvm/nvm.sh"
+          make react-init
+        fi
+      shell: bash -ileo pipefail {0}
```

Comment:
```suggestion
    - name: Restore pre-commit cache from cache
      id: cache-pre-commit-cache
      uses: actions/cache@v3
      with:
        path: |
          ~/.cache
        key: pre-commit-${{ hashFiles('**/.pre-commit-config.yaml') }}
    - name: Install pre-commit
      run: |
        pre-commit install-hooks
      shell: bash -ileo pipefail {0}
    - uses: actions/setup-node@v3
      with:
        node-version-file: '.nvmrc'
        cache: 'yarn'
        cache-dependency-path: '**/yarn.lock'
    - name: Initialize react
      run: make react-init
      shell: bash -ileo pipefail {0}
```

Reply:
Thank you for all the help & thorough review, it has been really helpful! 😃 
I've added your suggestion above with a couple minor edits